### PR TITLE
52 transforming knowledge

### PR DESCRIPTION
**Pull request by naina66406**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #<gh-issue-id>

Test URLs:
- Before: https://main--world-bank--aemsites.hlx.live/
- After: https://52-transforming-knowledge--world-bank--aemsites.hlx.live/en/drafts/naina/home

### CODE REVIEW

File: blocks/columns/columns.js [hunk 1]
```diff
@@ -3,6 +3,8 @@ import { div, ul, li } from '../../scripts/dom-helpers.js';
 export default function decorate(block) {
   const cols = [...block.firstElementChild.children];
   block.classList.add(`columns-${cols.length}-cols`);
+  const anchorContainer = document.createElement('div');
+  anchorContainer.className = 'anchor-container'; // Add a class for styling if needed
```
Comment (thomsebastin) [inline blocks/columns/columns.js:7]: Class should be added by author. Not dynamically like this. Also write columns logic if within a feature flag by checking if that class exists. This way current columns code is not overridden.

**Comment by naina66406** [comment]
Transforming knowledge section UI
image

File: blocks/columns/columns.js (hunk 1 above)
Comment (mkbansal1) [reply blocks/columns/columns.js:7]: I agree, i dont see any reason of customizing JS here. Just use variation and implement the rules in CSS

**Review by mkbansal1 — CHANGES_REQUESTED** [review]
On large screen size, image is going on right
image

File: blocks/columns/columns.css [hunk 2]
```diff
@@ -19,6 +19,57 @@
...
+      border-radius: 0;
+      padding: 0;
+      margin: 0;
+    }
+  }
+  .columns-img-col {
+    position: absolute;
+    right: 0;
+    top: -125px;
+    z-index: 1000;
```
Comment (thomsebastin) [inline blocks/columns/columns.css:69]: Why such large value? Lets use smaller z-index values. Easier to track.

File: blocks/columns/columns.css [hunk 3]
```diff
@@ -32,6 +83,42 @@
...
+        display: block;
+      }
+    }
+}
+  .bg-primary-blue-90 .columns-transform-teaser .columns-img-col picture {
+    width: 100%;
+  } 
+}
+
+@media(width <= 767px) {
```
Comment (thomsebastin) [inline blocks/columns/columns.css:103]: Use `768px` like everywhere else?

**Review by thomsebastin — CHANGES_REQUESTED** [review]

**Review by mkbansal1 — CHANGES_REQUESTED** [review]

File: blocks/columns/columns.css [hunk 4]
```diff
@@ -21,6 +21,65 @@
...
+}
+
+.columns.columns-transform-teaser > div p {
+  margin: 15px 0;
+}
+
+.columns.columns-divider-variant > div > div > p {
+  margin-top: 0;
+}
+
```
[the remark this answers is not in the document]
Comment (naina66406) [reply blocks/columns/columns.css:43]: I only moved it to fix specificity errors and it is not my change. Please check with Akshaya.

File: component-models.json [hunk 5]
```diff
@@ -135,6 +135,10 @@
               {
                 "name": "Feature Card Variation",
                 "value": "columns-feature-card"
+              },
+              {
+                "name": "transform-teaser",
```
[the remark this answers is not in the document]
Comment (naina66406) [reply component-models.json:140]: Done

File: blocks/columns/columns.css [hunk 6]
```diff
@@ -135,6 +190,26 @@
   padding-left: 10px;
 }
 
+@media(width<=1024px) {
+  .bg-primary-blue-90{
+    position: relative;
```
[the remark this answers is not in the document]
Comment (naina66406) [reply blocks/columns/columns.css:196]: Done

File: blocks/columns/columns.css [hunk 7]
```diff
@@ -21,6 +21,65 @@
...
+}
+
+.columns.columns-transform-teaser > div p:nth-child(3) a {
+  background-color: var(--color-white);
+  color: var(--primary-blue-100);
+  font-size: 16px;
+  margin: 0;
+}
+
+.columns.columns-transform-teaser > div p:nth-child(4) a[title="Search More Courses"] {
```
[the remark this answers is not in the document]
Comment (naina66406) [reply blocks/columns/columns.css:67]: Changed

File: icons/swish.svg
Comment (thomsebastin) [inline icons/swish.svg:1]: If you are not using this image anymore, then I suggest we remove it.

File: blocks/header/header.js [hunk 8]
```diff
@@ -155,7 +155,7 @@ function createSearchBox() {
     const searchInputContainer = div({ class: 'search-input-container' });
     const searchInputBox = document.createElement('input');
     const searchIcon = img({ class: 'search-icon' });
-    searchIcon.src = '../icons/search-white.svg';
+    searchIcon.src = '/icons/search-white.svg';
```
Comment (thomsebastin) [inline blocks/header/header.js:158]: Can you try `${window.hlx.codeBasePath}/icons/search-white.svg` and see if that works?

File: blocks/columns/columns.css [hunk 9]
```diff
@@ -188,6 +267,47 @@
...
 }
 
+@media (768px <= width <= 1024px) {
+  .columns.columns-transform-teaser .columns-img-col {
+    width: 50%;
+    right: -65px;
+  }
+}
+
+@media (width < 768px) {
```
Comment (thomsebastin) [inline blocks/columns/columns.css:277]: This is why we need to start mobile first. Can we not simplify by using
`width <= 1024px` & `width <= 768px`?

File: blocks/columns/columns.css [hunk 10]
```diff
@@ -135,6 +189,31 @@
...
+  .bg-primary-blue-90{
+    .columns.columns-transform-teaser  p{
+    &:nth-child(1){
+      font-size: 32px;
+      line-height: 35.2px;
+    }
+
+    &:nth-child(2){
+      font-size: 18px;
+      line-height: 25.2px;
```
Comment (thomsebastin) [inline blocks/columns/columns.css:202]: Can be rounded to `25`.

**Review by thomsebastin — CHANGES_REQUESTED** [review]

File: blocks/columns/columns.css (hunk 9 above)
Comment (naina66406) [reply blocks/columns/columns.css:276]: When using like this "width <= 1024px & width <= 768px"
its showing lint errors

File: blocks/columns/columns.css [hunk 11]
```diff
@@ -21,6 +21,64 @@
   order: 1;
 }
 
+.columns.columns-transform-teaser > div {
+  display: grid;
+  grid-template-columns: 45% 55%;
+  padding: 72px 0;
+  position: relative;
+
```
[the remark this answers is not in the document]
Comment (naina66406) [reply blocks/columns/columns.css]: Done

File: icons/swish.svg
Comment (naina66406) [reply icons/swish.svg:1]: Done

File: blocks/header/header.js (hunk 8 above)
Comment (naina66406) [reply blocks/header/header.js]: Its Working

File: blocks/columns/columns.css (hunk 10 above)
Comment (naina66406) [reply blocks/columns/columns.css]: Done

**Review by mkbansal1 — APPROVED** [review]

**Review by thomsebastin — APPROVED** [review]

**Pull request merged**